src/app/products/share-products/dividends-share-product/dividends.components.ts

```diff
@@ -0,0 +1,40 @@
+/**Angular Imports */
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatPaginator, MatSort, MatTableDataSource } from '@angular/material';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'mifosx-dividends-share-product',
+  templateUrl: './dividends.component.html',
+  styleUrls: ['./dividends.component.scss']
+})
+export class ShareProductsDividendsComponent implements OnInit {
+  dividendsData: any;
+  displayedColumns: string[] = ['name', 'startDividend', 'endDividend', 'status'];
+  dataSource: MatTableDataSource<any>;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+
+  constructor(private route: ActivatedRoute) {
+    this.route.data.subscribe((data: {dividends: any}) => {
+      this.dividendsData = data.dividends.pageItems;
+    });
+   }
+
+
+  ngOnInit() {
+    this.setDividends();
+
+  }
```

Comment:
```suggestion
    this.setDividends();
  }
  
```
Indentations are very important, many people will see your code when it's merged.